doc/long_vignettes/demes_event_timings.md

```diff
@@ -0,0 +1,181 @@
+---
+jupytext:
+  formats: md:myst
+  text_representation:
+    extension: .md
+    format_name: myst
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+(demes_event_timings)=
+
+# Timing of events in demographic models
+
+This document provides several examples showing *when* events happen in demographic models.
+
+We will define very simple setups that allow very simple assertions about their outcomes.
+While most simulations will involve more complex setups, the advantage of simplicity here
+is that we can easily validate the results because we can easily reason about the models.
+
+These examples come from the `fwdpy11` test suite.
+
+The key concept here is that the [demes specification](https://popsim-consortium.github.io/demes-spec-docs/main/introduction.html)
+defines an epoch as a half open interval on `[present, past)`.
+Our reasoning about the outcomes of data that we record during simulation involves breaking our models up into half open intervals and asking questions about them.
+
+Why do we spend time going over this?
+Imagine that you want to know the details about the ancestry of a given generation in your simulation.
+For example, the ancestry of the generation born during a pulse.
+One way to do that would be to preserve both the pulse generation and its parent generation as ancient samples.
+(See [here](recorders_vignette).)
+To do such sampling you need to know when to sample!
+
+## Some setup
+
+During some of our simulations, we will record the current time point
+and the part of the metadata containing individual parents.
+
+We can use a simple data class to record this information:
+
+
+```{literalinclude} ../../tests/test_demes_event_timings.py
+:language: python
+:lines: 8-20
```

Comment:
Not pointing out anything wrong.. but I suppose hard-coding line numbers can be dangerous if the source file changes as all?

Reply:
Yes, that is a problem.  Sadly, we just have t o be careful.